mapsforge-map-android/src/main/java/org/mapsforge/map/android/graphics/AndroidPaint.java

```diff
@@ -151,6 +156,90 @@ public int getTextWidth(String text) {
         return (int) this.paint.measureText(text);
     }
 
+    /**
+     * Uses StaticLayout to measure text accurately, and respects Spans.
+     * Returns width of the widest line.
+     */
+    @Override
+    public int getTextWidth(final String text, final int widthMax, boolean includePadding) {
+        int retVal = 0;
+
+        if (text != null && widthMax >= 0) {
+
+            final TextPaint myPaint;
+            if (this.paint instanceof TextPaint) {
+                myPaint = (TextPaint) this.paint;
+            } else {
+                myPaint = new TextPaint(this.paint);
+            }
+
+            final StaticLayout layout = new StaticLayout(text, myPaint, widthMax, Layout.Alignment.ALIGN_NORMAL, 1, 0, includePadding);
```

Comment:
Please add in the method:
```java
@SuppressWarnings("deprecation")
```

Reply:
Also it is better to use the `StaticLayout.Builder` for newer Androids.
See the implementation in `AndroidPointTextContainer`.